storage/Validate/validateartifactupload.groovy

```diff
@@ -0,0 +1,29 @@
+import org.artifactory.build.promotion.PromotionConfig
+import org.artifactory.build.staging.ModuleVersion
+import org.artifactory.build.staging.VcsConfig
+import org.artifactory.exception.CancelException
+import org.artifactory.repo.RepoPathFactory
+import org.artifactory.request.Request
+import org.artifactory.util.StringInputStream
+
+storage {
+      beforeCreate { item ->
+        def ff = item.isFolder() ? 'folder' : 'file'        
+        def repositorykey = item.repoPath.repoKey
+        def repositoryname ='repositoryName'
```

Comment:
A configuration file should be used to set the repositories list to be validated. 

Suggestion: 
- Create a json file named as the plugin with the following content:
```
["repositoryName1", "repositoryName2"]
```

- Use the snippet bellow as a reference to populate a list of repository names during plugin load:

```
def repositoryNames = new JsonSlurper().parse(new File(ctx.artifactoryHome.haAwareEtcDir, 'plugins/<NAME>.json'))
```

- Use this list to handle the events instead of having the repository name hard coded into the plugin

Check [restrictOverwrite](https://github.com/JFrogDev/artifactory-user-plugins/tree/master/storage/restrictOverwrite) and [mavenSnapshotCleanupWhenReleas](https://github.com/JFrogDev/artifactory-user-plugins/tree/master/cleanup/mavenSnapshotCleanupWhenRelease) for full examples.

Reply:
Thats a better way i assume so i completed it based on the standards in my new commit